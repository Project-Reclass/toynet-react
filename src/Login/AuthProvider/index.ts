import AuthProvider from './AuthProvider';
export * from './AuthProvider';
export { AuthActions } from './useAuthState';

export default AuthProvider;