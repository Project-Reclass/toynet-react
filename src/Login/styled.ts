import { Box, Input, Button } from '@chakra-ui/core';
import styled from '@emotion/styled';

export const Center = styled(Box)`
position: absolute;
left: 50%;
top: 50%;
transform: translate(-50%, -50%);
`;

export const ToyNetInput = styled(Input)`
border: none;
background-color: #212529;
`;

export const ToyNetButton = styled(Button)`
background-color: rgba(0,0,0,0);
border: 1pt solid white;

:hover {
  color: #212529;
}
`;
