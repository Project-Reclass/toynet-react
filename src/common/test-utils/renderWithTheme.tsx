import React, { FC } from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@chakra-ui/core/dist';
import renderer from 'react-test-renderer';

const Wrapper: FC = ({ children }) => (
  <ThemeProvider>{children}</ThemeProvider>
);

export const renderWithTheme = (ui: any) => {
  return render(ui, { wrapper: Wrapper });
};

export const renderTreeWithTheme = (ui: any) => {
  return renderer.create(<Wrapper>{ui}</Wrapper>);
};
