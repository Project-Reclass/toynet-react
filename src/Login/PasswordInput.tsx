import React, { FC, useState } from 'react';
import { InputGroup, InputProps, InputRightElement } from '@chakra-ui/core';

import { ToyNetButton, ToyNetInput } from './styled';

const PasswordInput: FC<React.PropsWithChildren<InputProps<HTMLInputElement>>> = ({ ...rest }) => {
  const [show, setShow] = useState(false);
  const handleClick = () => setShow(!show);

  return (
    <InputGroup size="md">
      <ToyNetInput
        pr="4.5rem"
        {...rest}
        type={show ? 'text' : 'password'}
        data-testid='password-input'
      />
      <InputRightElement width="4.5rem">
        <ToyNetButton h="1.75rem" size="sm" onClick={handleClick}>
          {show ? 'Hide' : 'Show'}
        </ToyNetButton>
      </InputRightElement>
    </InputGroup>
  );
};

export default PasswordInput;