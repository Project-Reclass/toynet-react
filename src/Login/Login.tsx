import React, { useEffect, useState } from 'react';
import {
  Box,
  Heading,
  Stack,
  useToast,
} from '@chakra-ui/core';

import { useAuthContext, AuthActions } from './AuthProvider';
import { useLogin } from 'src/common/api/login/hooks';
import { Center, ToyNetInput, ToyNetButton } from './styled';
import PasswordInput from './PasswordInput';

const Login = () => {
  const toast = useToast();
  const { dispatch } = useAuthContext();
  const [login, { data, isLoading, isSuccess, isError }] = useLogin();

  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  useEffect(() => {
    if (!isLoading && isSuccess && data) {
      dispatch({ type: AuthActions.LOGIN, payload: data });
      console.log('login success!', { user: data });
    }
  }, [data, dispatch, isLoading, isSuccess]);

  useEffect(() => {
    if (!isLoading && isError) {
      toast({
        title: 'Unable to log you',
        description: 'Please check your credentials and try again.',
        status: 'error',
        isClosable: true,
      });
    }
  }, [isError, isLoading, toast]);

  const handleEnter = (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (e.key === 'Enter')
      handleLogin();
  };

  const handleLogin = () => {
    if (username.length > 2 && password.length > 2) {
      login({ username, password });
      return;
    }
    toast({
      title: 'Invalid username or password.',
      description: 'Please enter a valid username and password',
      status: 'warning',
      isClosable: true,
    });
  };

  return (
    <Box width="100%" height="100vh">
      <Center width="25rem" height="30vh">
        <form>
        <Stack spacing={4}>
          <Heading textAlign='center'>Login to ToyNet!</Heading>
          <ToyNetInput
            placeholder='Username'
            type='text'
            autoComplete='username'
            onKeyPress={handleEnter}
            value={username}
            onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
              setUsername(e.currentTarget.value)}
          />
          <PasswordInput
            placeholder='Password'
            autoComplete='current-password'
            onKeyPress={handleEnter}
            value={password}
            onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
              setPassword(e.currentTarget.value)}
          />
          <ToyNetButton onClick={handleLogin}>
            Sign In
          </ToyNetButton>
        </Stack>
        </form>
      </Center>
    </Box>
  );
};

export default Login;