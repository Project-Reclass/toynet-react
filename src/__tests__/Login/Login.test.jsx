import React from 'react';

import { renderWithTheme } from 'src/common/test-utils/renderWithTheme';
import Login from 'src/Login';

describe('The Module List', () => {
  it('should render and match snapshot', () => {
    const { container } = renderWithTheme(<Login />);
    expect(container).toMatchSnapshot();
  });
});