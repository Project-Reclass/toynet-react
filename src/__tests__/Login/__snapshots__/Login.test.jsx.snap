// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Module List should render and match snapshot 1`] = `
<div>
  <div
    class="css-mikc7i"
  >
    <div
      class="css-1nlyd02"
    >
      <form>
        <div
          class="css-j7qwjs"
        >
          <h2
            class="css-1jwohdf"
          >
            Login to ToyNet!
          </h2>
          <input
            autocomplete="username"
            class="css-1kr1ry3"
            placeholder="Username"
            type="text"
            value=""
          />
          <div
            class="css-kjafn5"
          >
            <input
              autocomplete="current-password"
              class="css-a66fqy"
              data-testid="password-input"
              placeholder="Password"
              type="password"
              value=""
            />
            <div
              class="css-8ueujo"
            >
              <button
                class="css-13h65yf"
                type="button"
              >
                Show
              </button>
            </div>
          </div>
          <button
            class="css-u1m2zc"
            type="button"
          >
            Sign In
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
